A4A > Referrals: Redesign the A4A Referrals empty page

yashwin opened the pull request:
Resolves https://github.com/Automattic/jetpack-genesis/issues/342

## Proposed Changes

This PR redesigns the empty A4A Referrals page.

[Design](https://www.figma.com/design/fuufP6VNfZXYmvLsTRWRlG/Referrals?node-id=6711-122443&m=dev)

NOTE: 

- We have reused the existing component since we will get rid of the old UI very soon.
- `How does it work?` link doesn't work and will be fixed later
- The button `Add my details` will be the primary one, while the others will be secondary. Once the user fills in the data, the button becomes secondary, and the other buttons become primary.

## Testing Instructions

1. Open the A4A live link.
2. Append the /referrals URL as /referrals?flags=-a4a-automated-referrals, and verify that the above changes are not reflected and that the Referrals page matches the production.
3. Remove the feature flag > Go to  Referrals Dashboard & Verify the UI looks like below: 

<img width="1285" alt="Screenshot 2024-05-22 at 3 31 46 PM" src="https://github.com/Automattic/wp-calypso/assets/10586875/6aaf9738-06e5-4fc6-ac02-a36b4ba60ff7">

<img width="373" alt="Screenshot 2024-05-22 at 5 15 49 PM" src="https://github.com/Automattic/wp-calypso/assets/10586875/4a899d06-b98e-4afd-bfc4-fb3fd7ab47b4">



## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

yashwin committed:
Resdesign A4A Referrals empty page

madebynoam commented:
Looking great, @yashwin!

Just a few small notes:
Could we nudge the pill 1px up so that the baselines match:
<img width="369" alt="image" src="https://github.com/Automattic/wp-calypso/assets/8889977/fe752c78-ffa8-4b23-83ab-456a4ab85457">
![image](https://github.com/Automattic/wp-calypso/assets/8889977/fcd483c2-05ec-4718-96bf-e5bd918c5c9d)

At the small breakpoint, can we hide some of the title? So that the button text doesn't wrap:
<img width="423" alt="image" src="https://github.com/Automattic/wp-calypso/assets/8889977/f1e837e2-6f3f-48af-951c-79f8a3ebcaa7">

Otherwise, lgtm!

yashwin committed:
Address PR comments

yashwin commented:
> Looking great, @yashwin!

Thanks for the review, @madebynoam!

> Could we nudge the pill 1px up so that the baselines match: 

Makes sense. Updated:

<img width="579" alt="Screenshot 2024-05-22 at 5 22 01 PM" src="https://github.com/Automattic/wp-calypso/assets/10586875/5cee2489-e45b-41c2-861e-d10d96186dc4">

> At the small breakpoint, can we hide some of the title? So that the button text doesn't wrap:

Yes. Updated:

<img width="373" alt="Screenshot 2024-05-22 at 5 15 49 PM" src="https://github.com/Automattic/wp-calypso/assets/10586875/4224277f-a4cc-44b6-a586-8ac3d8aacf9a">

madebynoam commented:
> > Looking great, @yashwin!
> 
> Thanks for the review, @madebynoam!
> 
> > Could we nudge the pill 1px up so that the baselines match:
> 
> Makes sense. Updated:
> 
> <img alt="Screenshot 2024-05-22 at 5 22 01 PM" width="579" src="https://private-user-images.githubusercontent.com/10586875/332785012-5cee2489-e45b-41c2-861e-d10d96186dc4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTYzNzkyNTEsIm5iZiI6MTcxNjM3ODk1MSwicGF0aCI6Ii8xMDU4Njg3NS8zMzI3ODUwMTItNWNlZTI0ODktZTQ1Yi00MWMyLTg2MWUtZDEwZDk2MTg2ZGM0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTIyVDExNTU1MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY0MzkzMGJmZjM1YzMxMGVhNmUyYjRjYzNmYzc5NTFiYzNjMGEwNTU5M2U4YjFhNTRiZWQ5NDhjNDMxMWYyZjQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.TpCIeEfv-J6QTQxWsBZo0KM2q1ZspRyjWKphe2kvHKk">
> > At the small breakpoint, can we hide some of the title? So that the button text doesn't wrap:
> 
> Yes. Updated:
> 
> <img alt="Screenshot 2024-05-22 at 5 15 49 PM" width="373" src="https://private-user-images.githubusercontent.com/10586875/332785162-4224277f-a4cc-44b6-a586-8ac3d8aacf9a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTYzNzkyNTEsIm5iZiI6MTcxNjM3ODk1MSwicGF0aCI6Ii8xMDU4Njg3NS8zMzI3ODUxNjItNDIyNDI3N2YtYTRjYy00NGI2LWE1ODYtOGFjM2Q4YWFjZjlhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTIyVDExNTU1MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJlNDgxZWRhMTAyMTQyOWNjZjZlMzQ1OTI1YmU2ZjZkMDYzZjliNWJkYjk4NzhiOGEyYWU5YmZhNTMyMmU0NjImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.g24N5zxCGP2C6NOvEpFjqQjpbf4F6y7JDpP5p1JtxS4">

Thank you, @yashwin !

cleacos commented:
The background color for the buttons is the opposite; is it okay?

![image](https://github.com/Automattic/wp-calypso/assets/9832440/7e46f55a-830a-4638-bdb3-7d59f897ec6e)

cleacos reviewed (approved):
It works as it's expected. I left a minor comment about the button background color that doesn't match the screenshot in the description.

Other than that... LGTM!
